public/app/features/explore/extensions/ToolbarExtensionPoint.tsx

```diff
@@ -0,0 +1,127 @@
+import React, { lazy, ReactElement, Suspense, useMemo, useState } from 'react';
+
+import { type PluginExtensionLink, PluginExtensionPoints, RawTimeRange } from '@grafana/data';
+import { getPluginLinkExtensions } from '@grafana/runtime';
+import { DataQuery, TimeZone } from '@grafana/schema';
+import { Dropdown, Menu, ToolbarButton } from '@grafana/ui';
+import { contextSrv } from 'app/core/services/context_srv';
+import { truncateTitle } from 'app/features/plugins/extensions/utils';
+import { AccessControlAction, ExploreId, ExplorePanelData, useSelector } from 'app/types';
+
+import { getExploreItemSelector } from '../state/selectors';
+
+import { ConfirmNavigationModal } from './ConfirmNavigationModal';
+
+const AddToDashboard = lazy(() =>
+  import('./AddToDashboard').then(({ AddToDashboard }) => ({ default: AddToDashboard }))
+);
+
+type Props = {
+  exploreId: ExploreId;
+  timeZone: TimeZone;
+  splitted: boolean;
+};
+
+export function ToolbarExtensionPoint(props: Props): ReactElement | null {
+  const { exploreId, splitted } = props;
+  const [selectedExtension, setSelectedExtension] = useState<PluginExtensionLink | undefined>();
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+  const context = useExtensionPointContext(props);
+  const extensions = useExtensionLinks(context);
+  const selectExploreItem = getExploreItemSelector(exploreId);
+  const noQueriesInPane = useSelector(selectExploreItem)?.queries?.length;
+
+  // If we only have the explore core extension point registered we show the old way of
```

Comment:
I wonder if we could somehow reuse for "add to dashboard" extension here. It's not only small code duplication but also the title of the modal is not the same:

With no extensions and permissions only to add to an existing dashboard, we get a title saying "Add panel to existing dashboard":

https://github.com/grafana/grafana/assets/745532/4851b11c-a82b-4439-8291-79d0662e407d

But with multiple extensions, it's always set to "add panel to dashboard":

https://github.com/grafana/grafana/assets/745532/2a63dd8f-6d08-4353-a81e-1dead41f4c4f

Reply:
Great feedback! I will have a look at it. Didn't want to change too much of the current code but I'm happy to give it a go 🙌🏻